pkg/api/types.go

```diff
@@ -542,6 +542,59 @@ type ReplicationControllerList struct {
 	Items []ReplicationController `json:"items" yaml:"items"`
 }
 
+// PerNodeControllerSpec is the specification of a per-node pod controller.
+// As the internal representation of a per-node controller, it may have either
+// a TemplateRef or a Template set.
+type PerNodeControllerSpec struct {
+	//FIXME: how does it know whether it has already done a node or not?
+	//Maybe an automatic kubernetes.io/per-node-controller=uuid label?
+	// Selector is a label query over pods on each node that match this
+	// controller.
+	Selector map[string]string `json:"selector" yaml:"selector"`
+	//FIXME: need a node selector, too?
```

Comment:
NodeSelector is already in the pod template.

I think we will need something akin to per-attribute limits (no more than 1 per rack, per bus bar, etc.), but that doesn't need to be in the initial API.